### PR TITLE
Fix some warnings when running with valgrinds MPI wrapper.

### PR DESCRIPTION
**Pull request by blattms**
It warns whenever uninitialized data is sent. Please check whether the selected defaults are sound.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

File: opm/parser/eclipse/EclipseState/SimulationConfig/RockConfig.hpp [hunk 1]
```diff
@@ -89,8 +89,8 @@ struct RockComp {
     std::vector<RockComp> m_comp;
     std::string num_property = ParserKeywords::ROCKOPTS::TABLE_TYPE::defaultValue;
     std::size_t num_tables = ParserKeywords::ROCKCOMP::NTROCC::defaultValue;
-    bool m_water_compaction;
-    Hysteresis hyst_mode;
+    bool m_water_compaction = false;
+    Hysteresis hyst_mode = Hysteresis::NONE;
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/SimulationConfig/RockConfig.hpp:93]: For hysteresis the default is `Hysteresis::REVERS` so I guess that could be used.

File: src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp [hunk 2]
```diff
@@ -215,6 +215,9 @@ namespace Opm {
         using GC = ParserKeywords::GCOMPIDX;
         if (deck.hasKeyword<GC>())
             this->m_gas_comp_index = deck.getKeyword<GC>().getRecord(0).getItem<GC::GAS_COMPONENT_INDEX>().get<int>(0);
+        else
+            // Otherwise valgrind complains with MPI
+            this->m_gas_comp_index = 77;
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp:220]: Maybe rather set it in the `TableManager.hpp` file - a bit surprising this does not hit the other members of the `TableManager` class

**Review by joakim-hove — APPROVED** [review]

**Comment by blattms** [comment]
Thanks for the approval. Merging.

File: src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp (hunk 2 above)
Comment (blattms) [reply src/opm/parser/eclipse/EclipseState/Tables/TableManager.cpp:220]: Well they either must be set unconditionally or are not pods and hence default initialized.

**Comment by blattms** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Thanks for the updates - merging

**Pull request merged**